### PR TITLE
🐛 (games) Adding Handling Input Changes page

### PR DESCRIPTION
**Pull request by emilieroberts**
Adding Handling Input Changes page

Adding page about how apps can handle input device changes.

### CODE REVIEW

**Review by Snugug — CHANGES_REQUESTED** [review]
Please add `package.json` and `package-lock.json` back in

File: pages/en/games/handling-input-changes.md [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+You would like to add keyboard and bluetooth game controller support. Where do you begin?
+
+![Car racing game with touch controls](/images/games/optimizing-games-input/input-touch.png)
+
+#### Input states
+
+Begin by identifying all of the **input states** your game might be running in and then listing all the parameters that you would like to change in each state.
+
+<table>
```
Comment (Snugug) [inline pages/en/games/handling-input-changes.md:109]: Please write this table [using Markdown](https://www.markdownguide.org/extended-syntax/). You can use this [table generator](https://www.tablesgenerator.com/markdown_tables) as a starting place if you'd like

File: pages/en/games/handling-input-changes.md [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+If keyboard and touchscreen events were being received simultaneously, the game should stay in touchscreen mode - though it would still react to keyboard input. If no touchscreen input was received after 5 seconds and keyboard events were still being received, perhaps the on-screen controls would fade and the game would move to the keyboard state.
+
+Game controller input would follow a similar pattern: the controller UI state would be given a lower priority than keyboard/mouse and touch and only appear if game controller input and not other forms of input were being received. The game would always respond to controller input.
+
+Below is a state diagram and a transition table for the example. Adapt the idea to your app or game.
+
+![Prioritized state machine - touchscreen, keyboard/mouse, game controller](/images/games/optimizing-games-input/input-state-machine.png)
+
+<table>
```
Comment (Snugug) [inline pages/en/games/handling-input-changes.md:187]: Same with this table

File: pages/en/games/handling-input-changes.md [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+           return true // event handled here, return true
+       }
+   }
+   // If event not handled, pass up to system
+   return super.onKeyUp(keyCode, event)
+}
+```
+
+!!! aside.message--note
+**Note**: for KeyEvents, you can choose to use either [onKeyDown()](<https://developer.android.com/reference/android/app/Activity#onKeyDown(int,%20android.view.KeyEvent)>) or [onKeyUp()](<https://developer.android.com/reference/android/app/Activity#onKeyUp(int,%20android.view.KeyEvent)>). Here, [onKeyDown()](<https://developer.android.com/reference/android/app/Activity#onKeyDown(int,%20android.view.KeyEvent)>) is used for controlling the state machine whereas [onKeyUp()](<https://developer.android.com/reference/android/app/Activity#onKeyUp(int,%20android.view.KeyEvent)>) is used for triggering game events.
```
Comment (Snugug) [inline pages/en/games/handling-input-changes.md:297]: Can you please put these function calls in code ticks? Thanks!

**Review by Snugug — CHANGES_REQUESTED** [review]
A couple of small formatting items I noticed as well

**Review by fuentespatrick — APPROVED** [review]

**Comment by fuentespatrick** [comment]
Merging since @Snugug already reviewed and feedback was addressed

**Pull request merged**